python/kvikio/tests/test_async_io.py

```diff
@@ -29,8 +29,13 @@ def test_read_write(tmp_path, size):
     assert check_bit_flags(f.open_flags(), os.O_WRONLY)
     assert f.raw_write_async(a, stream.ptr).check_bytes_done() == a.nbytes
 
+    if f.is_compat_mode_preferred_for_async():
+        expected_except_msg = "Operation not permitted"
+    else:
+        expected_except_msg = "unsupported file open flags"
+
     # Try to read file opened in write-only mode
-    with pytest.raises(RuntimeError, match="Operation not permitted"):
+    with pytest.raises(RuntimeError, match=expected_except_msg):
```

Comment:
not a big fan of tests that check the error message content, but that's not on this PR.

Reply:
It see ms like we're adding a lot of public APIs just for this error. Can we instead just verify that the error matches _one_ of these strings, without worrying about which one? That would let us remove all the `is_compat_mode_preferred` logic.

Reply:
That makes sense. I'm wondering if ideally the public function `is_compat_mode_preferred()` and `is_compat_mode_preferred_for_async()` should be included in the Python API, or are they considered too low-level?

Reply:
The Python API addition has been reverted for now.